Subjet_BehavData_v5.py

```diff
@@ -1 +1,312 @@
+bID = 3025432
+sID = 658178
+encOnsets = pd.read_csv('Onset-Event-Encoding_CIMAQ_'+str(bID)+'*.txt', sep='\s+', header=None)
+encOnsets.columns = ["index", "Condi", "Trial", "Image", "Trial_seg", "onsetSec", "unknown"]
+mainFile = pd.read_csv('Output-Responses-Encoding_CIMAQ_'+bID+'*.txt', sep='\t')
+retriev = pd.read_csv('Output_Retrieval_CIMAQ_'+bID+'*.txt', sep='\t')
+
+#useful pandas basics:
+mainFile.dtypes #returns type of data inferred per column; make explicit?
+mainFile.head()
+mainFile.tail()
+mainFile.iloc[0:10, 0:10]
+mainFile.loc[:, 'TrialNumber']
+#to explicitely convert dtypes from one to another; return copy unless copy=False
+#convert subset of columns to dtype
+mainFile[['Category', 'OldNumber']]=mainFile[['Category', 'OldNumber']].astype(object) #string is object in pandas, str in Python
+#other types: bool (pandas and python), int64 (pandas) vs int (python), float64 (pandas) vs float (python)
+#string is object in pandas, str in Python
+
+###Qs for Samira: what purpose of first 3 trials? (signal equilibrium?)
+### What is the control task (what looks like?)
+###How are pictures labelled? Is the same image always old, or is
+###the oldness or newness of the image randomized between participants?
+###Can I have a directory with the stimulus images in it?
+##How are images labelled? (old - new, number always matches same pic?)
+##How to interpret response and response time to source question if post-scan pic identified as new?
+##(do they just press whatever?)
+
+#####################################
+
+#Analysis steps:
+
+#load main file: Output-Responses-Encoding_CIMAQ*txt, modify it in-place
+mainFile = pd.read_csv('Output-Responses-Encoding_CIMAQ_'+bID+'*.txt', sep='\t')
+
+#Modify main file in-place:
+#remove first three junk rows (blank trials): CTL0, Enc00 and ENc000
+mainFile.drop([0, 1, 2], axis=0, inplace=True)
+#alternative way: mainFile = mainFile.iloc[3:,]
+
+#Change column header names:
+#Column1: 'Category', change to "Condition"
+#Col3, OldNumber: change to "StimulusID"
+#Col6, Stim_RESP: change to "InScan_Resp";
+#Col7, Stim_RT: change to "InScan_RT", convert to float and change value to s (rather than ms)
+mainFile.rename(columns={'Category':'Condition', 'OldNumber':'StimulusID',
+'Stim_RESP':'InScan_Resp', 'Stim_RT':'InScan_RT'}, inplace=True)
+
+#cast RT data as float64, then divide by 1000 (from ms to s)
+mainFile[['InScan_RT']]=mainFile[['InScan_RT']].astype('float64', copy=False)
+mainFile['InScan_RT'] = mainFile['InScan_RT'].div(1000)
+
+#Remove Col8: 'Stim_ACC' (redundant)
+mainFile.drop('Stim_ACC', axis=1, inplace=True) #axis 0 = rows, 1 = columns
+
+#Load other enconding file with onset times (encOnsets): Onset-Event-Encoding_CIMAQ_*.txt
+##Data is Fixed Width, padding with blank spaces
+#Column format is estimated from first 100 rows by default; reset to avoid truncated values
+encOnsets = pd.read_fwf('Onset-Event-Encoding_CIMAQ_'+str(bID)+'*.txt', infer_nrows=210, delim_whitespace=True, header=None)
```

Comment:
Dummy question: Is this cautious to hard code infer_nrows ?

Reply:
If I don't, the default is 100, and it leads to some of my data being truncated in rows 200s+ (the trial numbers are missing their last digit)